File: app/src/main/java/com/my/televip/virtuals/ui/ProfileActivity.java
package com.my.televip.virtuals.ui;

import com.my.televip.obfuscate.Obfuscate;
import com.my.televip.virtuals.ActionBar.ActionBarMenuItem;

import de.robv.android.xposed.XposedHelpers;

public class ProfileActivity {

    Object profileActivity;

    public ProfileActivity(Object obj){
        profileActivity = obj;
    }

    public long getChatId(){
        return XposedHelpers.getLongField(profileActivity, Obfuscate.getFieldName("ProfileActivity", "chatId"));
    }

    public long getUserId(){
        return XposedHelpers.getLongField(profileActivity, Obfuscate.getFieldName("ProfileActivity", "userId"));
    }

    public Object[] getNameTextView(){
        return  (Object[]) XposedHelpers.getObjectField(profileActivity, Obfuscate.getFieldName("ProfileActivity", "nameTextView"));
    }

    public Object[] getOnlineTextView(){
        return  (Object[]) XposedHelpers.getObjectField(profileActivity, Obfuscate.getFieldName("ProfileActivity", "onlineTextView"));
    }

    public ActionBarMenuItem getOtherItem(){
        return new ActionBarMenuItem(XposedHelpers.getObjectField(profileActivity, Obfuscate.getFieldName("ProfileActivity", "otherItem")));
    }


    public BaseFragment getBaseFragment(){
        return new BaseFragment(profileActivity);
    }
}

File: app/src/main/java/com/my/televip/virtuals/ui/ProfileActivity.kt
package com.my.televip.virtuals.ui

import com.my.televip.obfuscate.Obfuscate
import com.my.televip.virtuals.ActionBar.ActionBarMenuItem
import de.robv.android.xposed.XposedHelpers

class ProfileActivity(@JvmField val profileActivity: Any?) {

    val userId: Long
        get() {
            val target = profileActivity ?: return 0L
            return XposedHelpers.getLongField(target, Obfuscate.getFieldName("ProfileActivity", "user_id"))
        }

    val chatId: Long
        get() {
            val target = profileActivity ?: return 0L
            return XposedHelpers.getLongField(target, Obfuscate.getFieldName("ProfileActivity", "chat_id"))
        }

    val nameTextView: Array<Any>?
        get() {
            val target = profileActivity ?: return null
            @Suppress("UNCHECKED_CAST")
            return XposedHelpers.getObjectField(target, Obfuscate.getFieldName("ProfileActivity", "nameTextView")) as? Array<Any>
        }

    val onlineTextView: Array<Any>?
        get() {
            val target = profileActivity ?: return null
            @Suppress("UNCHECKED_CAST")
            return XposedHelpers.getObjectField(target, Obfuscate.getFieldName("ProfileActivity", "onlineTextView")) as? Array<Any>
        }

    val otherItem: ActionBarMenuItem
        get() {
            val target = profileActivity ?: return ActionBarMenuItem(null)
            val result = XposedHelpers.getObjectField(target, Obfuscate.getFieldName("ProfileActivity", "otherItem"))
            return ActionBarMenuItem(result)
        }

    val baseFragment: BaseFragment?
        get() {
            val target = profileActivity ?: return null
            return BaseFragment(target)
        }
}
